internal/openstack/client.go

```diff
@@ -154,3 +165,31 @@ func (client *OpenStackClient) NewNetworkV2() (*gophercloud.ServiceClient, error
 		Region: client.region,
 	})
 }
+
+// Use the project name and project domain ID from the auth options to get the project ID.
+func getProjectID(provider *gophercloud.ProviderClient, cfg *AuthOpts) (string, error) {
+	identityClient, err := openstack.NewIdentityV3(provider, gophercloud.EndpointOpts{Region: cfg.Region})
+	if err != nil {
+		return "", err
+	}
+
+	projectPage, err := projects.List(
+		identityClient,
+		projects.ListOpts{DomainID: cfg.ProjectDomainID, Name: cfg.ProjectName},
+	).AllPages()
+	if err != nil {
+		return "", err
+	}
+
+	projectList, err := projects.ExtractProjects(projectPage)
+	if err != nil {
+		return "", err
+	}
+
+	// Project names are unique within their domain, so there should never be more than one result.
+	if len(projectList) < 1 {
+		return "", fmt.Errorf("no project named %s in domain %s", cfg.ProjectName, cfg.ProjectDomainID)
+	}
```

Comment:
Please also error out if there is more than one project. That should not happen when filtering for name + domain, and if it *does* happen, it is some kind of unexpected situation/bug and may lead to unexpected consequences. We should bail out instead.

I'll see how we can get this tested in the meantime.

Reply:
done